perf_test/sparse/KokkosSparse_spmv_mv.cpp

```diff
@@ -0,0 +1,452 @@
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <cstdio>
+
+#include <cstdlib>
+#include <ctime>
+#include <cstring>
+#include <cstdlib>
+#include <limits>
+#include <limits.h>
+#include <cmath>
+#include <unordered_map>
+
+#ifdef HAVE_CUSPARSE
+#include <cusparse.h>
+#endif
+
+#include <Kokkos_Core.hpp>
+#include <matrix_market.hpp>
+
+#include <KokkosKernels_SPMV.hpp>
+#include <Kokkos_SPMV.hpp>
+#include <Kokkos_SPMV_Inspector.hpp>
+#include <CuSparse_SPMV.hpp>
+#include <MKL_SPMV.hpp>
+
+#include <KokkosSparse_spmv.hpp>
+
+#ifdef _OPENMP
+#include <OpenMPStatic_SPMV.hpp>
+#include <OpenMPDynamic_SPMV.hpp>
+#include <OpenMPSmartStatic_SPMV.hpp>
+#endif
+
+enum {KOKKOS, MKL, CUSPARSE, KK_KERNELS, KK_KERNELS_INSP, KK_INSP, OMP_STATIC, OMP_DYNAMIC, OMP_INSP};
+enum {AUTO, DYNAMIC, STATIC};
+enum {REF, NEW};
+
+#ifdef INT64
+typedef long long int LocalOrdinalType;
+#else
+typedef int LocalOrdinalType;
+#endif
+
+
+template< typename ScalarType , typename OrdinalType>
+int SparseMatrix_generate(OrdinalType nrows, OrdinalType ncols, OrdinalType &nnz, OrdinalType varianz_nel_row, OrdinalType width_row, ScalarType* &values, OrdinalType* &rowPtr, OrdinalType* &colInd)
+{
+  rowPtr = new OrdinalType[nrows+1];
```

Comment:
@mhoemmen this would break compatibility with the current matrix reader interface and I do not wish to rewrite them now.
I will keep this in mind for when time can be spared to tackle a refactor of `SparseMatrix_MatrixMarket_read` and `SparseMatrix_ReadBinaryFormat` in order to have them support Kokkos views.